install-panel.sh

```diff
@@ -354,109 +444,90 @@ function install_pteroq {
   echo "* Installed pteroq!"
 }
 
-function create_database {
-  if [ "$OS" == "centos" ]; then
-    # secure MariaDB
-    echo "* MariaDB secure installation. The following are safe defaults."
-    echo "* Set root password? [Y/n] Y"
-    echo "* Remove anonymous users? [Y/n] Y"
-    echo "* Disallow root login remotely? [Y/n] Y"
-    echo "* Remove test database and access to it? [Y/n] Y"
-    echo "* Reload privilege tables now? [Y/n] Y"
-    echo "*"
-
-    mysql_secure_installation
-
-    echo "* The script should have asked you to set the MySQL root password earlier (not to be confused with the pterodactyl database user password)"
-    echo "* MySQL will now ask you to enter the password before each command."
-
-    echo "* Create MySQL user."
-    mysql -u root -p -e "CREATE USER '${MYSQL_USER}'@'127.0.0.1' IDENTIFIED BY '${MYSQL_PASSWORD}';"
-
-    echo "* Create database."
-    mysql -u root -p -e "CREATE DATABASE ${MYSQL_DB};"
-
-    echo "* Grant privileges."
-    mysql -u root -p -e "GRANT ALL PRIVILEGES ON ${MYSQL_DB}.* TO '${MYSQL_USER}'@'127.0.0.1' WITH GRANT OPTION;"
-
-    echo "* Flush privileges."
-    mysql -u root -p -e "FLUSH PRIVILEGES;"
-  else
-    echo "* Performing MySQL queries.."
-
-    echo "* Creating MySQL user.."
-    mysql -u root -e "CREATE USER '${MYSQL_USER}'@'127.0.0.1' IDENTIFIED BY '${MYSQL_PASSWORD}';"
+##################################
+# OS specific install functions ##
+##################################
 
-    echo "* Creating database.."
-    mysql -u root -e "CREATE DATABASE ${MYSQL_DB};"
+apt_update() {
+  apt update -q -y && apt upgrade -y
+}
 
-    echo "* Granting privileges.."
-    mysql -u root -e "GRANT ALL PRIVILEGES ON ${MYSQL_DB}.* TO '${MYSQL_USER}'@'127.0.0.1' WITH GRANT OPTION;"
+yum_update() {
+  yum -y update
+}
 
-    echo "* Flushing privileges.."
-    mysql -u root -e "FLUSH PRIVILEGES;"
+dnf_update() {
+  dnf -y upgrade
+}
 
-    echo "* MySQL database created & configured!"
-  fi
+enable_services_debian_based() {
+  systemctl enable mariadb
+  systemctl enable redis-server
+  systemctl start mariadb
+  systemctl start redis-server
 }
 
-##################################
-# OS specific install functions ##
-##################################
+enable_services_centos_based() {
+  systemctl enable mariadb
+  systemctl enable nginx
```

Comment:
Do we have to enable nginx on CentOS?

Reply:
Yes, I just tested, `dnf` or `yum` doesn't enable or start `nginx`

Reply:
It might have problems, because I need to reboot machine to have a working panel at least on CentOS 8 (also maybe on CentOS 7).

Reply:
Ok enabling might be the only thing needed for CentOS. I didn't get that pid error this time, so maybe it has been fixed.